src/transaction/offset.rs

```diff
@@ -41,7 +41,7 @@ impl Transaction {
     pub(crate) fn _input_coin_predicate_offset(&self, index: usize) -> Option<usize> {
         self.input_offset(index)
             .map(|ofs| ofs + Input::coin_predicate_offset())
-            .filter(|_| self.inputs()[index].is_coin())
+            .filter(|_| self.inputs()[index].is_coin_predicate())
```

Comment:
nit: the comment `...if this input is of type `CoinPredicate` ?

Reply:
Done! Lost the track of the git commit since it was rebased